collector/collector.go

```diff
@@ -196,7 +197,11 @@ func (c *Collector) addBuildAndJobMetrics(r *Result) error {
 				continue
 			}
 
-			pipeline := *build.Pipeline.Name
+			pipeline, ucError := idna.ToASCII(*build.Pipeline.Name)
+
+			if ucError != nil {
+				return false
```

Comment:
Again, not entirely sure what the idiomatic thing to do here is; I'm guessing false is a failure?

Reply:
`return ucError`

Reply:
That gives me
```
collector/collector.go:203:5: cannot use ucError (type error) as type bool in return argument
```

Reply:
Apologies, code reading comprehension fail! I'd go for:

```go
if usErr != nil {
   log.Printf("Error converting pipeline to ascii: %v", usErr)
   continue
}
```

Reply:
FWIW `err` is usually idiomatic in golang.

Reply:
I was just reading the code for better comprehension myself. I guess the other option would be to make the function passed into `Pages` return a `bool, error` instead of just a `bool` and handle it there. Then we can get what I _think_ is the desired effect of a crash-out rather than just losing some metrics?

Reply:
It absolutely could, but it probably should continue to process subsequent pipelines, which is why errors are handled elsewhere with `continue`.

Reply:
Ah, so we'd prefer to just lose some metrics (which I guess is what will happen if we just log and `continue`? Forgive my ignorance if `continue` does something special to do with error handling in Go). Will amend now!

Reply:
Yup, that was generally my thinking. `continue` is just the usual for loop construct that will continue to the next loop iteration!